packages/grid/_modules_/grid/components/columnHeaders/ColumnHeaderMenuIcon.tsx

```diff
@@ -24,12 +24,7 @@ export function ColumnHeaderMenuIcon(props: ColumnHeaderFilterIconProps) {
     (event: React.MouseEvent<HTMLButtonElement>) => {
       event.preventDefault();
       event.stopPropagation();
-      const lastMenuState = apiRef!.current.getState().columnMenu;
-      if (!lastMenuState.open || lastMenuState.field !== column.field) {
-        apiRef!.current.showColumnMenu(column.field, columnMenuId, columnMenuButtonId);
-      } else {
-        apiRef!.current.hideColumnMenu();
-      }
+      apiRef!.current.toggleColumnMenu(column.field, columnMenuId, columnMenuButtonId);
```

Comment:
if you click on another 3 dots icon the menu will blink

Reply:
No, it doesn't (or at least I can't see it). What `toggleColumnMenu ` does is what I removed. I just moved that block of code into a separate method.